shared-bindings/wifi/Radio.c

```diff
@@ -163,6 +163,86 @@ STATIC mp_obj_t wifi_radio_connect(size_t n_args, const mp_obj_t *pos_args, mp_m
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_KW(wifi_radio_connect_obj, 1, wifi_radio_connect);
 
+//|     ap_rssi: int
+//|     """RSSI of the currently connected AP. Returns none if not connected"""
+//|
+STATIC mp_obj_t wifi_radio_get_ap_rssi(mp_obj_t self) {
+    return common_hal_wifi_radio_get_ap_rssi(self);
+
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_get_ap_rssi_obj, wifi_radio_get_ap_rssi);
+
+const mp_obj_property_t wifi_radio_ap_rssi_obj = {
+    .base.type = &mp_type_property,
+    .proxy = { (mp_obj_t)&wifi_radio_get_ap_rssi_obj,
+               (mp_obj_t)&mp_const_none_obj,
+               (mp_obj_t)&mp_const_none_obj },
+};
+
+//|     ap_ssid: int
+//|     """SSID of the currently connected AP. Returns none if not connected"""
+//|
+STATIC mp_obj_t wifi_radio_get_ap_ssid(mp_obj_t self) {
+    return common_hal_wifi_radio_get_ap_ssid(self);
+
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_get_ap_ssid_obj, wifi_radio_get_ap_ssid);
+
+const mp_obj_property_t wifi_radio_ap_ssid_obj = {
+    .base.type = &mp_type_property,
+    .proxy = { (mp_obj_t)&wifi_radio_get_ap_ssid_obj,
+               (mp_obj_t)&mp_const_none_obj,
+               (mp_obj_t)&mp_const_none_obj },
+};
+
+//|     ap_bssid: int
+//|     """BSSID (usually MAC) of the currently connected AP. Returns none if not connected"""
+//|
+STATIC mp_obj_t wifi_radio_get_ap_bssid(mp_obj_t self) {
+    return common_hal_wifi_radio_get_ap_bssid(self);
+
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_get_ap_bssid_obj, wifi_radio_get_ap_bssid);
+
+const mp_obj_property_t wifi_radio_ap_bssid_obj = {
+    .base.type = &mp_type_property,
+    .proxy = { (mp_obj_t)&wifi_radio_get_ap_bssid_obj,
+               (mp_obj_t)&mp_const_none_obj,
+               (mp_obj_t)&mp_const_none_obj },
+};
+
+//|     ipv4_gateway: Optional[ipaddress.IPv4Address]
```

Comment:
Do these three properties need to be ipv4 specific? Could they just be `gateway`, `subnet` and `dns` and either return IPv4 or IPv6 addresses? Maybe we'd need to return more than one object then?

Reply:
Gateway and subnet are coming from the same ```esp_netif_ip_info_t``` that you got the ip address from so they're always going to be ipv4.  I just matched the naming scheme you used for the existing ```ipv4_address``` property honestly - it's easy enough to drop the prefixes, but I do think it's nice when everything was grouped together when you dir() the ```wifi.radio``` object.

For DNS, though, esp-idf's DNS type (```esp_netif_dns_info_t```) contains both ```esp_ip6_addr_t ip6``` and ```esp_ip4_addr_t```, so I was trying to be a little more clear on which one I was picking out.

Reply:
Ok, I think the prefix is fine since it's possible to have both ipv4 and ipv6 versions of these.